dialer/bandit.go

```diff
@@ -114,6 +114,23 @@ func (o *BanditDialer) chooseDialerForDomain(network, addr string) (ProxyDialer,
 	notAllFailing := hasNotFailing(o.dialers)
 	for i := 0; i < (len(o.dialers) * 2); i++ {
 		d = o.dialers[chosenArm]
+		readyChan := d.Ready()
+		if readyChan != nil {
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Since this is critical path for the user, let's just see if the dialer is ready _right now_ rather than wait for a timeout.  So:
```
select {
case err := <-readyChan:
    // Check and handle err
default:
    // Dialer is not ready, call differentArm
}
```